/retest
Bazel-test failed with this:
```
FAIL: //pkg/scheduler:go_default_test (run 2 of 2) (see /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/pkg/scheduler/go_default_test/run_2_of_2/test.log)
INFO: From Testing //pkg/scheduler:go_default_test (run 2 of 2):
==================== Test output for //pkg/scheduler:go_default_test (run 2 of 2):
--- FAIL: TestCreateFromConfig (0.09s)
    --- FAIL: TestCreateFromConfig/policy_with_unspecified_predicates_or_priorities_uses_default (0.04s)
        factory_test.go:374: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            + 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 1}},
              	{Name: "NodeAffinity", Args: &config.NodeAffinityArgs{}},
              	... // 4 identical elements
              }
    --- FAIL: TestCreateFromConfig/policy_with_arguments (0.03s)
        factory_test.go:374: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            + 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 1}},
              	{Name: "NodeAffinity", Args: &config.NodeAffinityArgs{}},
              	... // 3 identical elements
              }
    --- FAIL: TestCreateFromConfig/policy_with_HardPodAffinitySymmetricWeight_argument (0.01s)
        factory_test.go:374: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            + 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 10}},
              	{Name: "NodeResourcesFit", Args: &config.NodeResourcesFitArgs{}},
              } 
```

but adding the defaultpreemption args causes it to fail locally for me:
```
$ go test -run TestCreateFromConfig ./pkg/scheduler/...
--- FAIL: TestCreateFromConfig (0.00s)
    --- FAIL: TestCreateFromConfig/policy_with_unspecified_predicates_or_priorities_uses_default (0.00s)
        factory_test.go:387: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            - 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 1}},
              	{Name: "NodeAffinity", Args: &config.NodeAffinityArgs{}},
              	... // 4 identical elements
              }
    --- FAIL: TestCreateFromConfig/policy_with_arguments (0.00s)
        factory_test.go:387: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            - 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 1}},
              	{Name: "NodeAffinity", Args: &config.NodeAffinityArgs{}},
              	... // 3 identical elements
              }
    --- FAIL: TestCreateFromConfig/policy_with_HardPodAffinitySymmetricWeight_argument (0.00s)
        factory_test.go:387: unexpected plugins config diff (-want, +got):   []config.PluginConfig{
            - 	{Name: "DefaultPreemption", Args: s"&TypeMeta{Kind:,APIVersion:,}"},
              	{Name: "InterPodAffinity", Args: &config.InterPodAffinityArgs{HardPodAffinityWeight: 10}},
              	{Name: "NodeResourcesFit", Args: &config.NodeResourcesFitArgs{}},
              }
FAIL
FAIL	k8s.io/kubernetes/pkg/scheduler	0.410s
```

--- a/cmd/kube-scheduler/app/options/configfile.go
+++ b/cmd/kube-scheduler/app/options/configfile.go
@@ -19,6 +19,7 @@ package options
 import (
 	"fmt"
 	"io/ioutil"
+	"k8s.io/klog/v2"
 	"os"
 
 	"k8s.io/apimachinery/pkg/runtime"
@@ -48,8 +49,13 @@ func loadConfig(data []byte) (*kubeschedulerconfig.KubeSchedulerConfiguration, e
 	return nil, fmt.Errorf("couldn't decode as KubeSchedulerConfiguration, got %s: ", gvk)
 }
 
-// WriteConfigFile writes the config into the given file name as YAML.
-func WriteConfigFile(fileName string, cfg *kubeschedulerconfig.KubeSchedulerConfiguration) error {
+// LogOrWriteConfig logs the completed component config and writes it into the given file name as YAML, if either is enabled
+func LogOrWriteConfig(fileName string, cfg *kubeschedulerconfig.KubeSchedulerConfiguration, completedProfiles []kubeschedulerconfig.KubeSchedulerProfile) error {
+	if !(klog.V(2).Enabled() || len(fileName) > 0) {
+		return nil
+	}
+	cfg.Profiles = completedProfiles
+
 	const mediaType = runtime.ContentTypeYAML
 	info, ok := runtime.SerializerInfoForMediaType(kubeschedulerscheme.Codecs.SupportedMediaTypes(), mediaType)
 	if !ok {
@@ -57,15 +63,26 @@ func WriteConfigFile(fileName string, cfg *kubeschedulerconfig.KubeSchedulerConf
 	}
 
 	encoder := kubeschedulerscheme.Codecs.EncoderForVersion(info.Serializer, kubeschedulerconfigv1beta1.SchemeGroupVersion)
-
-	configFile, err := os.Create(fileName)
-	if err != nil {
-		return err
+	if klog.V(2).Enabled() {
+		bytes, err := runtime.Encode(encoder, cfg)
+		if err != nil {
+			return err
+		}
+		configString := string(bytes)
+		klog.Infof("Using component config:\n%+v\n", configString)
 	}
-	defer configFile.Close()
-	if err := encoder.Encode(cfg, configFile); err != nil {
-		return err
-	}
 
+	if len(fileName) > 0 {
+		configFile, err := os.Create(fileName)
+		if err != nil {
+			return err
+		}
+		defer configFile.Close()
+		if err := encoder.Encode(cfg, configFile); err != nil {
+			return err
+		}
+		klog.Infof("Wrote configuration to: %s\n", fileName)
+		os.Exit(0)
+	}
 	return nil
 }

--- a/cmd/kube-scheduler/app/server.go
+++ b/cmd/kube-scheduler/app/server.go
@@ -128,14 +128,6 @@ func runCommand(cmd *cobra.Command, opts *options.Options, registryOptions ...Op
 		return err
 	}
 
-	if len(opts.WriteConfigTo) > 0 {
-		if err := options.WriteConfigFile(opts.WriteConfigTo, &cc.ComponentConfig); err != nil {
-			return err
-		}
-		klog.Infof("Wrote configuration to: %s\n", opts.WriteConfigTo)
-		return nil
-	}
-
 	return Run(ctx, cc, sched)
 }
 
@@ -310,6 +302,7 @@ func Setup(ctx context.Context, opts *options.Options, outOfTreeRegistryOptions 
 	}
 
 	recorderFactory := getRecorderFactory(&cc)
+	completedProfiles := make([]kubeschedulerconfig.KubeSchedulerProfile, 0)
 	// Create the scheduler.
 	sched, err := scheduler.New(cc.Client,
 		cc.InformerFactory,
@@ -323,8 +316,15 @@ func Setup(ctx context.Context, opts *options.Options, outOfTreeRegistryOptions 
 		scheduler.WithPodInitialBackoffSeconds(cc.ComponentConfig.PodInitialBackoffSeconds),
 		scheduler.WithExtenders(cc.ComponentConfig.Extenders...),
 		scheduler.WithParallelism(cc.ComponentConfig.Parallelism),
+		scheduler.WithBuildFrameworkCapturer(func(profile kubeschedulerconfig.KubeSchedulerProfile) {
+			// Profiles are processed during Framework instantiation to set default plugins and configurations. Capturing them for logging
+			completedProfiles = append(completedProfiles, profile)
+		}),
 	)
 	if err != nil {
+		return nil, nil, err
+	}
+	if err := options.LogOrWriteConfig(opts.WriteConfigTo, &cc.ComponentConfig, completedProfiles); err != nil {
 		return nil, nil, err
 	}
 

--- a/pkg/scheduler/BUILD
+++ b/pkg/scheduler/BUILD
@@ -65,6 +65,7 @@ go_test(
         "//pkg/scheduler/framework:go_default_library",
         "//pkg/scheduler/framework/plugins:go_default_library",
         "//pkg/scheduler/framework/plugins/defaultbinder:go_default_library",
+        "//pkg/scheduler/framework/plugins/defaultpreemption:go_default_library",
         "//pkg/scheduler/framework/plugins/interpodaffinity:go_default_library",
         "//pkg/scheduler/framework/plugins/nodeaffinity:go_default_library",
         "//pkg/scheduler/framework/plugins/nodelabel:go_default_library",

--- a/pkg/scheduler/factory_test.go
+++ b/pkg/scheduler/factory_test.go
@@ -38,6 +38,7 @@ import (
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/apis/config"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 	frameworkplugins "k8s.io/kubernetes/pkg/scheduler/framework/plugins"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/interpodaffinity"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeaffinity"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodelabel"
@@ -107,6 +108,13 @@ func TestCreateFromConfig(t *testing.T) {
 				"apiVersion" : "v1"
 			}`),
 			wantPluginConfig: []schedulerapi.PluginConfig{
+				{
+					Name: defaultpreemption.Name,
+					Args: &schedulerapi.DefaultPreemptionArgs{
+						MinCandidateNodesPercentage: 10,
+						MinCandidateNodesAbsolute:   100,
+					},
+				},
 				{
 					Name: interpodaffinity.Name,
 					Args: &schedulerapi.InterPodAffinityArgs{
@@ -232,6 +240,13 @@ func TestCreateFromConfig(t *testing.T) {
 			}`),
 			wantPluginConfig: []schedulerapi.PluginConfig{
 				{
+					Name: defaultpreemption.Name,
+					Args: &schedulerapi.DefaultPreemptionArgs{
+						MinCandidateNodesPercentage: 10,
+						MinCandidateNodesAbsolute:   100,
+					},
+				},
+				{
 					Name: interpodaffinity.Name,
 					Args: &schedulerapi.InterPodAffinityArgs{
 						HardPodAffinityWeight: 1,
@@ -312,6 +327,13 @@ func TestCreateFromConfig(t *testing.T) {
 				"hardPodAffinitySymmetricWeight" : 10
 			}`),
 			wantPluginConfig: []schedulerapi.PluginConfig{
+				{
+					Name: defaultpreemption.Name,
+					Args: &schedulerapi.DefaultPreemptionArgs{
+						MinCandidateNodesPercentage: 10,
+						MinCandidateNodesAbsolute:   100,
+					},
+				},
 				{
 					Name: interpodaffinity.Name,
 					Args: &schedulerapi.InterPodAffinityArgs{

